Balance/Profile/PasswordUpdateView.swift

```diff
@@ -19,41 +22,90 @@ struct PasswordUpdateView: View {
-                        )
-                        .padding(5)
-                        .padding(.horizontal, 25)
+                    titleView
+                    passwordFields
                     Spacer()
                     saveButton
                 }
             }
+            .onAppear(perform: listen)
+            .onChange(of: authModel.authError) { value in
+                if !value.isEmpty {
+                    if value == "OK" {
+                        self.alertMessage = "Password saved successfully"
+                        password = ""
+                        confirmPassword = ""
+                        UIApplication.shared.sendAction(#selector(UIResponder.resignFirstResponder), to: nil, from: nil, for: nil)
+                    } else {
+                        self.alertMessage = value
+                    }
+                    self.showingAlert = true
+                }
+            }
+            .alert(alertMessage, isPresented: $showingAlert) {
+                Button("OK", role: .cancel) {
+                    authModel.authError = ""
+                }
+            }
+        }
+    }
+    
+    var passwordFields: some View {
+        Group {
+            ShowHideSecureField("New password", text: $password)
+                .padding()
+                .font(.custom("Montserrat-Regular", size: 17))
+                .foregroundColor(darkGrayColor)
+                .overlay(
+                    RoundedRectangle(cornerRadius: 14)
+                        .stroke(lightGrayColor, lineWidth: 1)
+                )
+                .padding(5)
+                .padding(.horizontal, 25)
+            ShowHideSecureField("Confirm password", text: $confirmPassword)
+                .padding()
+                .font(.custom("Montserrat-Regular", size: 17))
+                .foregroundColor(darkGrayColor)
+                .overlay(
+                    RoundedRectangle(cornerRadius: 14)
+                        .stroke(lightGrayColor, lineWidth: 1)
+                )
+                .padding(5)
+                .padding(.horizontal, 25)
         }
     }
     
+    var titleView: some View {
+        Text("Password update: ")
+            .font(.custom("Nunito-Bold", size: 18))
+            .foregroundColor(darkBlueColor)
+            .frame(maxWidth: .infinity, alignment: .leading)
+            .padding(20)
+    }
+    
     var saveButton: some View {
         Button(action: {
-            dismiss()
+                if $password.wrappedValue.isEmpty ||
+                    $confirmPassword.wrappedValue.isEmpty {
+                    self.alertMessage = "All the information is required"
+                    showingAlert = true
+                    return
+                }
+                
+                if $password.wrappedValue !=
+                    $confirmPassword.wrappedValue {
+                    self.alertMessage = "The confirmation password mismatch"
```

Comment:
I would use either `The confirmation password did not match` or `Confirmation password mismatch` @gperissetcelteeka

Reply:
ok